src/Thruway/Role/Subscriber.php

```diff
@@ -219,13 +251,49 @@ public function subscribe(ClientSession $session, $topicName, callable $callback
             "options"    => $options,
             "deferred"   => $deferred
         ];
-
+        
         array_push($this->subscriptions, $subscription);
-
+        
         $subscribeMsg = new SubscribeMessage($requestId, $options, $topicName);
         $session->sendMessage($subscribeMsg);
 
         return $deferred->promise();
     }
-
+	
+    /**
+     * process unsubscribe
+     * @param ClientSession $session
+     * @param string $subscriptionId
+     * @return boolean
+     */
+    public function unsubscribe(ClientSession $session, $subscriptionId)
+    {
+        $requestId = Utils::getUniqueId();
+        $subscriptionExists = false;
+        $deferred  = new Deferred();
+        
+        foreach ($this->subscriptions as $i => $subscription) {
+            if ($subscription["subscription_id"] == $subscriptionId) {
+                $subscriptionExists = true;
+                
+                $this->subscriptions[$i]["unsubscribed_request_id"] = $requestId;
+                $this->subscriptions[$i]["unsubscribed_deferred"] = $deferred;
+            }
+        }
+        
+        // In case the client never subscribed to this topic before
+        if ($subscriptionExists === false) {
+            $unsubscriptionPromise = [
```

Comment:
You're right. This way, the `unsubscriptionPromises` attribute I created for this class became useless, so I got rid of it.